substrate/frame/revive/src/exec.rs

```diff
@@ -965,12 +972,13 @@ where
 	/// Run the current (top) frame.
 	///
 	/// This can be either a call or an instantiate.
-	fn run(&mut self, executable: E, input_data: Vec<u8>) -> ExecResult {
+	fn run(&mut self, executable: E, input_data: Vec<u8>) -> Result<(), ExecError> {
 		let frame = self.top_frame();
 		let entry_point = frame.entry_point;
 		let delegated_code_hash =
 			if frame.delegate_caller.is_some() { Some(*executable.code_hash()) } else { None };
 
+		self.take_caller_output();
```

Comment:
I assume we want to drop early for memory reasons? Needs a comment. I guess this is possible if we make `frame` a reference to a mutable frame.

```suggestion
		frame.last_frame_output = Default::default();
```

Reply:
I agree a comment would be appropriate here, added it. However we want to drop the caller frame here, but `self.top_frame_mut()` would reference the top frame (callee).